graylog2-server/src/main/java/org/graylog/events/processor/DBEventDefinitionService.java

```diff
@@ -64,15 +66,26 @@ public EventDefinitionDto saveWithOwnership(EventDefinitionDto eventDefinitionDt
         return dto;
     }
 
-    @Override
-    public int delete(String id) {
+    public int deleteUnregister(String id) {
+        // Must ensure mutability before deleting, so that de-registration is only performed if entity exists
+        // and is not mutable.
+        ensureMutability(get(id).orElseThrow(() -> new IllegalArgumentException("Event Definition not found.")));
+        return doDeleteUnregister(id, () -> super.delete(id));
+    }
+
+    public int deleteUnregisterImmutable(String id) {
+        return doDeleteUnregister(id, () -> super.deleteImmutable(id));
+    }
+
```

Comment:
@danotorrey @kingzacko1 I just noticed this while looking at Dan's other PR but I think it's more relevant here and probably needs another "core" PR. 

If you have an Event Definition installed by Illuminate with a Notification associated with it, when uninstalling the pack the Notifications fail because `NotificationResourceHandler.delete()` tries to update any associated Event Definitions to remove the Notification, which fails because the Event Definitions are Immutable. https://github.com/Graylog2/graylog2-server/blob/a9771993b18ee8ce3c0bdecca9a20afe1ad612e0/graylog2-server/src/main/java/org/graylog/events/notifications/NotificationResourceHandler.java#L180

I'm not sure how much of an edge case this is, and whether the other entities need it, but I think it might be good to have a `saveImmutable()` method here or probably the superclass for situations like this. What do y'all think?

```
java.lang.IllegalArgumentException: Immutable entity cannot be modified
	at org.graylog2.database.entities.ScopedDbService.ensureMutability(ScopedDbService.java:80)
	at org.graylog2.database.entities.ScopedDbService.save(ScopedDbService.java:66)
	at org.graylog.events.notifications.NotificationResourceHandler.lambda$delete$4(NotificationResourceHandler.java:180)
	at com.google.common.collect.ImmutableList.forEach(ImmutableList.java:422)
	at org.graylog.events.notifications.NotificationResourceHandler.delete(NotificationResourceHandler.java:170)
	at org.graylog.events.contentpack.facade.NotificationFacade.delete(NotificationFacade.java:113)
	at org.graylog.events.contentpack.facade.NotificationFacade.delete(NotificationFacade.java:47)
	at org.graylog2.contentpacks.ContentPackService.uninstallContentPack(ContentPackService.java:286)
	at org.graylog2.contentpacks.ContentPackService.uninstallContentPack(ContentPackService.java:237)
	at org.graylog.plugins.illuminate.installation.IlluminateContentPackInstaller.lambda$null$2(IlluminateContentPackInstaller.java:74)
	at java.lang.Iterable.forEach(Iterable.java:75)
	at org.graylog.plugins.illuminate.installation.IlluminateContentPackInstaller.lambda$uninstallAllRevisions$3(IlluminateContentPackInstaller.java:74)
	at java.lang.Iterable.forEach(Iterable.java:75)
	at org.graylog.plugins.illuminate.installation.IlluminateContentPackInstaller.uninstallAllRevisions(IlluminateContentPackInstaller.java:72)
	at org.graylog.plugins.illuminate.installation.BundleInstaller.uninstallContentPacksForSpotlights(BundleInstaller.java:260)
	at org.graylog.plugins.illuminate.installation.BundleInstaller.uninstallContentPacksForSpotlightIds(BundleInstaller.java:254)
	at org.graylog.plugins.illuminate.bundle.BundleManagementService.lambda$uninstallPacksForBundle$10(BundleManagementService.java:391)
	at org.graylog.plugins.illuminate.bundle.BundleManagementService.suspendProcessingAndExecute(BundleManagementService.java:356)
	at org.graylog.plugins.illuminate.bundle.BundleManagementService.uninstallPacksForBundle(BundleManagementService.java:387)
	at org.graylog.plugins.illuminate.rest.BundleManagementResource.managePacks(BundleManagementResource.java:175)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
```

Reply:
Since it sounds like there won't be notifications on Illuminate Event Defs and a user will only be adding Notifications to mutable copies, I don't think this is really necessary. Unless y'all can think of another reason.

Reply:
Thanks for testing and looking out @ryan-carroll-graylog! I agree, sounds like it's not much of an issue.